crates/bevy_render/src/camera/camera.rs

```diff
@@ -138,6 +142,74 @@ impl Camera {
             None
         }
     }
+
+    /// Given a position in screen space, compute the world-space line that corresponds to it.
+    pub fn screen_to_world_ray(
+        &self,
+        pos_screen: Vec2,
+        windows: &Windows,
+        images: &Assets<Image>,
+        camera_transform: &GlobalTransform,
+    ) -> Line {
+        let camera_position = camera_transform.compute_matrix();
+        let window_size = self.target.get_logical_size(windows, images).unwrap();
+        let projection_matrix = self.projection_matrix;
+
+        // Normalized device coordinate cursor position from (-1, -1, -1) to (1, 1, 1)
+        let cursor_ndc = (pos_screen / window_size) * 2.0 - Vec2::from([1.0, 1.0]);
+        let cursor_pos_ndc_near: Vec3 = cursor_ndc.extend(-1.0);
+        let cursor_pos_ndc_far: Vec3 = cursor_ndc.extend(1.0);
+
+        // Use near and far ndc points to generate a ray in world space
+        // This method is more robust than using the location of the camera as the start of
+        // the ray, because ortho cameras have a focal point at infinity!
+        let ndc_to_world: Mat4 = camera_position * projection_matrix.inverse();
+        let cursor_pos_near: Vec3 = ndc_to_world.project_point3(cursor_pos_ndc_near);
+        let cursor_pos_far: Vec3 = ndc_to_world.project_point3(cursor_pos_ndc_far);
```

Comment:
I'm not certain if the non-projection matrix can be combined with the projection matrix like that without the perspective divide happening before transforming by the view matrix. Could you compare the results with the following?
```suggestion
        let inverse_projection = projection_matrix.inverse();
        let cursor_pos_view_near = inverse_projection.project_point3(cursor_pos_ndc_near);
        let cursor_pos_view_far = inverse_projection.project_point3(cursor_pos_ndc_far);
        let cursor_pos_near = view_matrix.transform_point3(cursor_pos_view_near);
        let cursor_pos_far = view_matrix.transform_point3(cursor_pos_view_far);
```